### ISSUE TITLE
Add a batch normalization layer with its back propagation

### ISSUE THREAD

**Issue by zishansami102** [zishansami102/CNN-from-Scratch#6]
[pointer](https://github.com/zishansami102/CNN-from-Scratch/blob/master/MNIST/convnet.py#L72-L84)
This is a little vague pointer as it will depend on where you are trying to put the batch_norm layer.
Everything depends on what is working and what is not and what is the conventional way to it.

**Comment by AdityaSoni19031997** [comment]
Basically you want to add batch norm ?

**Comment by zishansami102** [comment]
Yes

**Comment by AdityaSoni19031997** [comment]
I can help in adding the batch_norm layer but not sure about its back_prop part..
Can someone help?

**Comment by zishansami102** [comment]
[cs231n Backpropagation](https://youtu.be/i94OvYb6noo)
[A little implementation part of the same](http://cs231n.github.io/optimization-2/)

Above links would help you to implement backpropagation with any computation graph(In this case it is Batch_Norm)

**Comment by AdityaSoni19031997** [comment]
The assignments of CS231N have the batch_norm setup for CIFAR 10

**Comment by zishansami102** [comment]
Backnorm? you mean batch_norm? or back_prop?
either one!! None of it depends on the type of dataset. Back_prop depends on Computation Graph and Batch_norm depends on direct input image.

**Comment by AdityaSoni19031997** [comment]
That's true that it doesn't depend on the dataset but atleast on the arch...
In our case the arch is fix..

**Comment by AdityaSoni19031997** [comment]
Any improvement?

**Comment by AdityaSoni19031997** [comment]
Hi, I have implemented the forward and the backwards pass of batch Norm layer in python using numpy but it's in function format, Will it help?
Will try to incorporate it into you code style
**Is this what you want?**
![image](https://user-images.githubusercontent.com/22738086/38107064-7fbb5972-33ae-11e8-8404-04c8cce65da0.png)

**Comment by zishansami102** [comment]
Yes, you can integrate that into the convnet() function.

**Comment by AdityaSoni19031997** [comment]
Will try my best
